goss-testing/automated/ncn-storage-checks

```diff
@@ -27,91 +27,31 @@
 
 export GOSS_LOG_BASE_DIR=/opt/cray/tests/install/logs
 
-# Temporary file to store command output
-NCN_STORAGE_CHECKS_TMPFILE=/tmp/.ncn-storage-checks.tmp.$$.$RANDOM
-
 source "$GOSS_BASE/automated/run-ncn-tests.sh"
 
 echo $'\e[1;33m'Storage Node Automated Tests$'\e[0m'
 echo $'\e[1;33m'----------------------------$'\e[0m'
 
-# find the storage node names and query the server endpoint
-for FAKELOOP in 1 
-do
-    if ! command -v ceph &> /dev/null
-    then
-        echo "ceph command not found"
-    else
-        echo "ceph command found"
-        if ! ceph status -f json-pretty > ${NCN_STORAGE_CHECKS_TMPFILE} 2>&1
-        then
-            echo "Command failed: ceph status -f json-pretty"
-        elif ! mons=$(jq -r '.quorum_names[]' ${NCN_STORAGE_CHECKS_TMPFILE})
-        then
-            echo "Unexpected format from command: ceph status -f json-pretty"
-        else
-            # Make sure we got at least one NCN name
-            ncns=( $mons )
-            if [[ ${#ncns[@]} -gt 0 ]]
-            then
-                echo "Obtained storage node name list from: ceph status"
-                break
-            fi
-            echo "Unable to find node names in output of ceph status -f json-pretty"
-        fi
-    fi
-
-    if ! command -v kubectl &> /dev/null
-    then
-        echo "kubectl command not found"
-    else
-        echo "kubectl command found"
-        if ! kubectl get cm ceph-csi-config -o json > ${NCN_STORAGE_CHECKS_TMPFILE} 2>&1
-        then
-            echo "Command failed: kubectl get cm ceph-csi-config -o json"
-        elif ! jq -r '.data[]' ${NCN_STORAGE_CHECKS_TMPFILE} > ${NCN_STORAGE_CHECKS_TMPFILE}.2 2>&1
-        then
-            echo "Unexpected format from command: kubectl get cm ceph-csi-config -o json"
-        elif ! jq -r '.[].monitors[]' ${NCN_STORAGE_CHECKS_TMPFILE}.2 > ${NCN_STORAGE_CHECKS_TMPFILE} 2>&1
-        then
-            echo "Unexpected 'data' field format in output of command: kubectl get cm ceph-csi-config -o json"
-        else
-            # Make sure we got at least one NCN name
-            ncns=( $mons )
-            if [[ ${#ncns[@]} -gt 0 ]]
-            then
-                echo "Obtained storage node name list from: kubectl get cm ceph-csi-config"
-                break
-            fi
-            echo "Unable to find node names in output of kubectl get cm ceph-csi-config -o json"
-        fi
-    fi
-    
-    echo "Using default list of storage NCNs"
-    mons="ncn-s001 ncn-s002 ncn-s003"
-done
-echo -n "List of storage NCNs: "
-echo $mons
-
-# Remove the temporary file(s), if they exist
-rm -f ${NCN_STORAGE_CHECKS_TMPFILE} ${NCN_STORAGE_CHECKS_TMPFILE}.2 >& /dev/null
-
-(( counter=0 ))
-
-for node in $mons; do
-    nodename=$(echo $node|cut -d":" -f1)
-    echo "$counter: Checking $nodename"
-    if [[ $(nc -z -w 10 $node 22) ]] || [[ $counter -lt 3 ]]
```

Comment:
Do we know that the netcat command is not needed anymore to verify that the storage node is alive and responding?

Reply:
Good question. I think we're okay here - we basically cover this implicitly with this PR, because we try to run the goss tests on all storage nodes. For any which aren't up (or not on the network, anyway), we'll get a failure in trying to run the goss tests.